src/diffusers/models/modeling_utils.py

```diff
@@ -458,18 +458,35 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 " dispatching. Please make sure to set `low_cpu_mem_usage=True`."
             )
 
+        # Load config if we don't provide a configuration
+        config_path = pretrained_model_name_or_path
+
         user_agent = {
             "diffusers": __version__,
             "file_type": "model",
             "framework": "pytorch",
         }
 
-        # Load config if we don't provide a configuration
-        config_path = pretrained_model_name_or_path
+        # load config
+        config, unused_kwargs = cls.load_config(
+            config_path,
+            cache_dir=cache_dir,
+            return_unused_kwargs=True,
+            force_download=force_download,
+            resume_download=resume_download,
+            proxies=proxies,
+            local_files_only=local_files_only,
+            use_auth_token=use_auth_token,
+            revision=revision,
+            subfolder=subfolder,
+            device_map=device_map,
+            user_agent=user_agent,
+            **kwargs,
+        )
+        _commit_hash = config.pop("_commit_hash", None)
```

Comment:
NIT: I think commit_hash as a variable name shouldn't be prefixed with an underscore. An underscore prefix as idiomatic python usually means an unused variable. Understandable the key has to be underscore prefixed in the config but I don't think there's much confusion around `commit_hash = config.pop('_commit_hash')`

Reply:
Fair point